sdk/storage/Azure.Storage.DataMovement.Files.Shares/tests/ShareDestinationCheckpointDataTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Text;
+using Azure.Storage.Files.Shares.Models;
+using Azure.Storage.Test;
+using NUnit.Framework;
+using Metadata = System.Collections.Generic.IDictionary<string, string>;
+
+namespace Azure.Storage.DataMovement.Files.Shares.Tests
+{
+    public class ShareDestinationCheckpointDataTests
+    {
+        private const string DefaultContentType = "text/plain";
+        private readonly string[] DefaultContentEncoding = new string[] { "gzip" };
+        private readonly string[] DefaultContentLanguage = new string[] { "en-US" };
+        private const string DefaultContentDisposition = "inline";
+        private const string DefaultCacheControl = "no-cache";
+        private readonly Metadata DefaultMetadata = DataProvider.BuildMetadata();
+
+        private ShareFileDestinationCheckpointData CreateDefault()
+        {
+            return new ShareFileDestinationCheckpointData(
+                new ShareFileHttpHeaders()
+                {
+                    ContentType = DefaultContentType,
+                    ContentEncoding = DefaultContentEncoding,
+                    ContentLanguage = DefaultContentLanguage,
+                    ContentDisposition = DefaultContentDisposition,
+                    CacheControl = DefaultCacheControl,
+                },
+                DefaultMetadata);
+        }
+
+        private void AssertEquals(ShareFileDestinationCheckpointData left, ShareFileDestinationCheckpointData right)
+        {
+            Assert.That(left.Version, Is.EqualTo(right.Version));
```

Comment:
nit: Any reason to use `Assert.That` and `Is.EqualTo` over just `Assert.AreEqual`?

Reply:
`Assert.That` was introduced in nunit 3.0 and is what their docs recommend over `Assert.AreEqual` from previous major revs that they just kept for ease of upgrading.

Advantages include:
- Reads like english for fast test comprehension
- Properly integrates what the old model separates into `CollectionAssert`
- Is generally more flexible/complex in what it can check while still maintaining its layout, rather than requiring several additional lines/branches
- Harder to accidentally write `Assert.Equals` which is incorrect

Reply:
Ok cool, did not know that was the recommendation. I will start switching to this format.